core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -842,6 +847,75 @@ class ControllerApis(val requestChannel: RequestChannel,
       }
   }
 
+  def handleCreateDelegationTokenRequest(request: RequestChannel.Request): CompletableFuture[Unit] = {
+    val alterRequest = request.body[CreateDelegationTokenRequest]
+
+    val requester = request.context.principal
+    val ownerPrincipalName = alterRequest.data.ownerPrincipalName
+    val ownerPrincipalType = alterRequest.data.ownerPrincipalType
+    val owner = if (ownerPrincipalName == null || ownerPrincipalName.isEmpty) {
+      request.context.principal
+    } else {
+      new KafkaPrincipal(ownerPrincipalType, ownerPrincipalName)
+    }
+
+    // Requester is always allowed to create token for self
+    if (!owner.equals(requester) && 
```

Comment:
I agree. then we should also add `allowTokenRequests(request)` check to all controller APIs?

Reply:
I added the code. Please look over the added code. 

One change is that I had to allow PLAINTEXT because all our broker to controller forwarding tests use PLAINTEXT.